src/java.base/windows/classes/sun/nio/fs/WindowsFileCopy.java

```diff
@@ -174,7 +183,9 @@ static void copy(final WindowsPath source,
 
         // Use CopyFileEx if the file is not a directory or junction
         if (!sourceAttrs.isDirectory() && !sourceAttrs.isDirectoryLink()) {
-            final int flags = (!followLinks) ? COPY_FILE_COPY_SYMLINK : 0;
+            final int flags = ((!followLinks) ? COPY_FILE_COPY_SYMLINK : 0) |
+                              ((sourceAttrs.size() > UNBUFFERED_IO_THRESHOLD) ?
+                                  COPY_FILE_NO_BUFFERING : 0);
```

Comment:
Would this be easier to read:

```
        int flags = (followLinks) ? 0 : COPY_FILE_COPY_SYMLINK;
        if (sourceAttrs.size() > UNBUFFERED_IO_THRESHOLD) {
            flags |= COPY_FILE_NO_BUFFERING;
        }
```

Reply:
I did not do that because `flags` is final but I guess it need not be.

Reply:
Addressed in 82bbc8b779befc254093239897e25e5db85e4e0b.

Reply:
I think this update is much more readable - thanks.